fs/dafsclient/rmtfile.cpp

```diff
@@ -1789,12 +1789,14 @@ extern bool clientAsyncCopyFileSection(const char *uuid,
                         offset_t fromOfs,
                         offset_t size,
                         ICopyFileProgress *progress,
-                        unsigned timeout)       // returns true when done
+                        unsigned timeout,
+                        CFflags copyFlags)       // returns true when done
 {
     CRemoteFile *cfile = QUERYINTERFACE(from,CRemoteFile);
-    if (!cfile) {
-        // local - do sync
-        from->copySection(to,toOfs,fromOfs,size,progress);
+    if (!cfile || to.isLocal()) {
```

Comment:
curious if the addition of 'to.isLocal()' was actually necessary? i.e. when and by what, for a local file did a CRemoteFile get created?

Reply:
This was triggered when I enabled copyPhysical for the roxie copies.  The files were being copied from a remote cluster to this cluster - where the filenames are local.
Without this check the code executes the copy command on the remote dafilesrv, which then tries to connect to the dafilesrv on the k8s cluster - and then fails.   If there was a dafilesrv running on the cluster it would probably have worked (if the firewall allowed)..  For that reason alone it is better to pull than push.